src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,108 @@
-export const Pagination = () => {};
+import classNames from 'classnames';
+import { MouseEvent, useMemo } from 'react';
+
+type Props = {
+  total: number,
+  perPage: number,
+  currentPage: number,
+  onPageChange: (page: number) => void,
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const numberOfPages = useMemo(() => {
+    return Math.ceil(total / perPage);
+  }, [total, perPage]);
+
+  const handlePageClick = (event: MouseEvent) => {
+    event.preventDefault();
+    const target = event.target as HTMLElement;
+    const pageNumberAttr = target.getAttribute('data-page-number');
+    const pageNumber = pageNumberAttr ? +pageNumberAttr : 1;
+
+    onPageChange(pageNumber);
+  };
+
+  const pageItems = useMemo(() => {
+    const items = [];
+
+    for (let pageNumber = 1; pageNumber <= numberOfPages; pageNumber += 1) {
+      items.push(
+        <li
+          key={pageNumber}
+          className={classNames('page-item', {
+            active: currentPage === pageNumber,
+          })}
+        >
+          <a
+            data-cy="pageLink"
+            className="page-link"
+            href={`#${pageNumber}`}
+            data-page-number={pageNumber}
+            onClick={handlePageClick}
+          >
+            {pageNumber}
+          </a>
+        </li>,
+      );
+    }
+
+    return items;
+  }, [total, perPage, currentPage]);
```

Comment:
it is still not fixed. Let's generate an array of `pageNumbers` based on `total `and `perPage`. Then render this HTML with other HTML code by iterating through the `pageNumbers `array